### DISCUSSION TITLE
Subselects in cross environment update?

### DISCUSSION THREAD

**Discussion by FH-Inway** [MarkMpn/Sql4Cds#137] [Q&A] [answered]
Hello, taking my first steps with Sql4Cds these days, great tool, it already saved me a ton of work.

I was wondering if it is possible to write SQL statements where data in one environment is update with data from another environment? I was hoping since subqueries are supported, this would be possible. But I'm getting syntax errors for something like

``` sql
update [Dev].dbo.account
set field = (select field from [Uat].dbo.account where [Dev].dbo.account.accountnumber = [Uat].dbo.account.accountnumber)
```

Error: Unknown column: [Uat].dbo.account.accountnumber

**Comment by MarkMpn** [comment, marked as answer]
I haven't tested this with subqueries, but you should be able to rewrite this using a join:

```sql
UPDATE dev
SET    field = uat.field
FROM   Dev.dbo.account AS dev
       INNER JOIN Uat.dbo.account AS uat
       ON dev.accountnumber = uat.accountnumber
```

**Reply by FH-Inway** [reply]
Works perfect, thanks!

Man, almost 3.000 accounts updated in a minute. I was previously working with Excel export/import and workflows. Felt like stone age, now I'm in the 21st century :)

Again, thanks for an awesome tool.
